src/Employer/Employer.Web/Controllers/DashboardController.cs

```diff
@@ -43,6 +44,34 @@ public async Task<IActionResult> Dashboard([FromRoute] string employerAccountId,
             return View(vm);
         }
 
+        private void SaveFiltersInCookie(string filter, string search)
+        {
+            var value = JsonConvert.SerializeObject(new FilterCookie(filter, search));
+            Response.Cookies.SetSessionCookie(_hostingEnvironment, CookieNames.VacanciesFilter, value);
+        }
+
+        private void TryGetFiltersFromCookie(out string filter, out string search)
+        {
+            filter = string.Empty;
+            search = string.Empty;
+            var cookieValue = Request.Cookies.GetCookie(CookieNames.VacanciesFilter);
+            if (string.IsNullOrWhiteSpace(cookieValue)) return;
+            var values = JsonConvert.DeserializeObject<FilterCookie>(cookieValue);
```

Comment:
You've changed the value of the Filter cookie from a string to a json object.  If anyone is mid session  during the deployment and has a filter set we're going to try and desearialize a string E.g. `"Draft"` into a `FilterCookie` object.  I think we need to try/catch around the deserialization for this scenario.

Reply:
The change is fine - but you need to handle the failing of the deserialization as this cookie is already in play with a different data type. We don't have this problem with Provider as this cookie has always been a json object.

Reply:
👍